PetGame/Services/PetService.cs

```diff
@@ -200,5 +211,196 @@ public bool DeletePet(ulong petId, ulong ownerId)
+            }
+            else
+            {
+                //check the type of activity and subtract percentage accordingly
+                foreach (Activity Activity in PastActivities)
+                {
+                    //if a Feeding activity has occured, increase hunger
+                    if (Activity.Type.Equals(ActivityType.Feeding))
+                    {
+                        if (hungerPercentage < 1.0 &&
+                            ((hungerPercentage += HungerIncrease) <= 1.0))
+                        {
+                            hungerPercentage += HungerIncrease;
+                        }
+                        else
+                        {
+                            hungerPercentage = 1.0;
+                        }
+                    }
+                    //if a Training event has occured, subtract
+                    else if (Activity.Type.Equals(ActivityType.Training))
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= TrainingDecrease) > 0))
+                        {
+                            hungerPercentage -= TrainingDecrease;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
+                    }
+                    //if a Race event has occurred
+                    else if (Activity.Type.Equals(ActivityType.Race))
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= RaceDecrease) > 0))
+                        {
+                            hungerPercentage -= RaceDecrease;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
+                    }
+                    //else, decrease hunger
+                    else
+                    {
+                        if (hungerPercentage > 0 &&
+                            ((hungerPercentage -= HungerDecreasePerHour) > 0))
+                        {
+                            hungerPercentage -= HungerDecreasePerHour;
+                        }
+                        else
+                        {
+                            hungerPercentage = 0;
+                        }
+                    }
+                }
+            }//end
+
+            //check the number of hours since the last activity
+            int HoursSinceLastActivity = 0;
+            int MinutesTillNextAction = 5;
+
+            //check to see if there were any activities in the specified interval
+            if (PastActivities.Count == 0)
+            {
+                HoursSinceLastActivity = HoursToCheck;
+            }
+            else
+            {
+                //get the last activity
+                Activity LastActivity = PastActivities[PastActivities.Count - 1];
```

Comment:
The list of `PastActivities` is sorted by most recent to least recent. You'll always end up getting the activity that's the furthest away from the current time.